rpcs3/Emu/RSX/rsx_methods.h

```diff
@@ -665,16 +665,37 @@ namespace rsx
 			return decode<NV4097_SET_STENCIL_TEST_ENABLE>().stencil_test_enabled();
 		}
 
-		bool restart_index_enabled() const
+		u8 index_array_location() const
 		{
-			return decode<NV4097_SET_RESTART_INDEX_ENABLE>().restart_index_enabled();
+			return decode<NV4097_SET_INDEX_ARRAY_DMA>().index_dma();
+		}
+
+		rsx::index_array_type index_type() const
+		{
+			return decode<NV4097_SET_INDEX_ARRAY_DMA>().type();
 		}
 
 		u32 restart_index() const
 		{
 			return decode<NV4097_SET_RESTART_INDEX>().restart_index();
 		}
 
+		bool restart_index_enabled_raw() const
+		{
+			return decode<NV4097_SET_RESTART_INDEX_ENABLE>().restart_index_enabled();
+		}
+
+		bool restart_index_enabled() const
+		{
+			if (const u32 idx = restart_index(); 
```

Comment:
Games will usually swing entirely one way or the other and the invalid primitive restart index case is going to be practically non-existent since you cannot render models that are designed with restart without it. In this case, we're even doing index checks when restart index is not even enabled.

Reply:
I think it is potentially faster to process these elements with the assumption that a restart index won't be there.
A good example is that the processing of disjointed primitive's indices hardly can be vectorized because you may need to skip writing another value when a restart index is encountered.

Reply:
I'm not complaining about that, I'm saying the index check may want to be ignored when restarting is not enabled. If a game isn't using restart, it will never have the check make sense, its just wastage. i.e `if (!primitive_restart_enabled_raw()) return false` then do the other stuff

Reply:
ah ok, i see what you mean.